src/loop/typer.ml

```diff
@@ -238,6 +242,8 @@ module Make(S : State_intf.Typer) = struct
     | x :: [] -> Format.fprintf fmt "%d" x
     | x :: r -> Format.fprintf fmt "%d or %a" x print_expected r
 
+  type 'a T.err += Generic_type_error of string
```

Comment:
To be clear, I don't particularly like this change, but I recognize it could greatly simplify things for users of dolmen, so it's okay and should be kept in this PR/MR, I'm just wondering about your point of view as a user, ^^

Reply:
Can you give more detail about the context / piece of code where these are raised in colibri ?
In a builtin symbols typechecker extension, you should have all the required things (there are helper functions: `Tff._error` to help with constructing errors).